### ISSUE TITLE
memory leak when using semi data

### ISSUE THREAD

**Issue by ishaybe** [trypag/torchmed#6]
when I try to use semi train function I the memory usage of RAM is increasing without stopping.
do you know what is the problem?

**Comment by trypag** [comment]
No, that could be a loaded tensor not properly detached from the grad
history.
